src/SuperGoodRobot/SuperGoodRobot.ino

```diff
@@ -0,0 +1,447 @@
+		break;
+	case IR:
+		break;
+	case ZIPLINE:
+		break;
+	case FINISHED:
+		break;
+	case TEST:
+		break;
+	case CLAW_ARM_TEST:
+		start_claw_arm_test();
+		break;
+	case MAIN_MENU:
+		start_main_menu();
+		break;
+	case STATE_MENU:
+		start_state_menu();
+		break;
+	case TAPE_MENU:
+		start_tape_menu();
+		break;
+	case DRIVE:
+		start_followtape();
+		break;
+  }
+}
+
+void exitState(byte byteRobotState)
+{
+	RobotState state = (RobotState)byteRobotState;
+	switch (state)
+	{
+	case INITIALISING:
+		break;
+	case PAST_DOOR:
+		break;
+	case FOLLOW_TAPE_1: //nothing, really
+		break;
+	case COLLECT_ITEM_1:
+		break;
+	case FOLLOW_TAPE_2: //nothing, really
+		break;
+	case COLLECT_ITEM_2:
+		break;
+	case COLLECT_ITEM_3:
+		break;
+	case COLLECT_ITEM_4:
+		break;
+	case COLLECT_ITEM_5:
+		break;
+	case COLLECT_ITEM_6:
+		break;
+	case UP_RAMP:
+		break;
+	case PAST_RAMP:
+		break;
+	case IR:
+		break;
+	case ZIPLINE:
+		break;
+	case FINISHED:
+		break;
+	case TEST:
+		break;
+	case CLAW_ARM_TEST:
+		break;
+	case DRIVE:
+		break;
+	case MAIN_MENU:
+		break;
+	case STATE_MENU:
+		break;
+	case TAPE_MENU:
+		break;
+  }
+}
+
+//============== nice print/string functions ==============
+String byteStateToString(byte byteRobotState) {
+	RobotState stringRobotState = (RobotState)byteRobotState;
+	switch(stringRobotState) {
+	case INITIALISING:
+		return("INITIALISING");
+	case PAST_DOOR:
+		return("PAST DOOR");
+	case FOLLOW_TAPE_1:
+		return("TAPE 1");
+	case COLLECT_ITEM_1:
+		return("ITEM 1");
+	case FOLLOW_TAPE_2:
+		return("TAPE 2");
+	case COLLECT_ITEM_2:
+		return("ITEM 2");
+	case COLLECT_ITEM_3:
+		return("ITEM 3");
+	case COLLECT_ITEM_4:
+		return("ITEM 4");
+	case COLLECT_ITEM_5:
+		return("ITEM 5");
+	case COLLECT_ITEM_6:
+		return("ITEM 6");
+	case UP_RAMP:
+		return("UP RAMP");
+	case PAST_RAMP:
+		return("PAST RAMP");
+	case IR:
+		return("IR");
+	case ZIPLINE:
+		return("ZIPLINE");
+	case FINISHED:
+		return("FINISHED");
+	case TEST:
+		return("TESTING");
+	case CLAW_ARM_TEST:
+		return("CLAW ARM TEST");
+	case DRIVE:
+		return("TEST DRIVE");
+	case MAIN_MENU:
+		return("MAIN_MENU");
+	case STATE_MENU:
+		return("STATE MENU");
+	case TAPE_MENU:
+		return("TAPE MENU");
+	}
+}
+
+void LCDStateChangePrint(byte byteRobotState) {
+	String stringRobotState=byteStateToString(byteRobotState);
+	LCD.clear();
+	LCD.home();
+	LCD.setCursor(0,0);
+	LCD.print("SWITCH:");
+	LCD.setCursor(0,1);
+	LCD.print(stringRobotState);
+}
+
+String bytePlanToString(byte byte_Plan){
+	Plan stringPlan = (Plan)byte_Plan;
+	switch(stringPlan) {
+	case FULL:
+		return("FULL");
+	case NOT4:
+		return("ALL BUT 4");
+	case NOT5:
+		return("ALL BUT 5");
+	case TEST_DRIVE:
+		return("TEST DRIVE");
+  }
+}
+
+//============== SET POSITION/CHECK FUNCTIONS ==============
+void setForearmServo(int angle) {
+	RCServo0.write(angle);
+}
+
+void setBasearmServo(int angle) {
+	RCServo1.write(angle);
+}
+
+void setBaseServo(int angle) {
+	RCServo2.write(angle);
+}
```

Comment:
I'm all for abstraction, but how necessary are these?  It's hard to believe that these will ever be more than one liners.  Why not just use the function, instead of calling a function to call a function?